### PR TITLE
[MB-12283] ShipmentInfoList and ShipmentDetails bugs

### PR DESCRIPTION
**Pull request by SirenaBorracha**
## Summary

This PR adds missing functionality to the HHG `ShipmentInfoList` to allow the display accordion to drop down and show the complete record. It also includes a fix for an issue in `ShipmentDetails`, a parent component, to allow Storybook/Happo to properly load.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

Locally
1. Access the Office UI
2. Login as a TOO (I used `too_role@office.mil`)
3. Search for Move Code `HGNTSR`
4. On the Move Details page under the Approved Shipments heading, click the caret icons to the right of the HHG and NTSR moves and see that they expand and collapse the accordion showing the correct fields when collapsed and and expanded respectively (see this [Slack comment](https://ustcdp3.slack.com/archives/CP4979J0G/p1657146426215509?thread_ts=1657136724.161959&cid=CP4979J0G) that outlines which are which)
5. Choose another few examples from the dashboard and confirm that the caret successfully expands and collapses

Storybook
1. Search for the Shipment Details component or scroll down to the entry below the heading Office Components
2. The component should load without the error complaining that `moveCode` is `undefined` (see this [Slack post](https://ustcdp3.slack.com/archives/CP4979J0G/p1657037860419829) referring to the error)
3. View the Default version and the 'Without Permissions' version to see the edit buttons disappear without permissions 

## Verification Steps for Author

These are to be checked by the author.

- [x] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [x] Request review from a member of a different team.
- [x] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: src/components/Office/DefinitionLists/ShipmentInfoList.jsx [hunk 1]
```diff
@@ -122,11 +126,11 @@ const ShipmentInfoList = ({ className, shipment, warnIfMissing, errorIfMissing,
     >
       {requestedPickupDateElement}
       {pickupAddressElement}
-      {secondaryPickupAddress && secondaryPickupAddressElement}
+      {showElement(secondaryPickupAddressElementFlags) && secondaryPickupAddressElement}
```
Comment (YanZ777) [inline src/components/Office/DefinitionLists/ShipmentInfoList.jsx:129]: **thought:** It looks like `showElement(secondaryPickupAddressElementFlags)` might be returning false, which is why the tests can't find the element.

File: src/components/Office/DefinitionLists/ShipmentInfoList.test.jsx [hunk 2]
```diff
@@ -93,11 +93,15 @@ describe('Shipment Info List', () => {
       }),
     ).toBeInTheDocument();
 
-    const releasingAgent = screen.getByText(labels.agents[0]);
-    expect(within(releasingAgent.parentElement).getByText(info.agents[0].email, { exact: false })).toBeInTheDocument();
+    const releasingAgent = screen.getByText(labels.mtoAgents[0]);
```
Comment (YanZ777) [inline src/components/Office/DefinitionLists/ShipmentInfoList.test.jsx:96]: **suggestion:** Tests should be added to check that the button to expand and collapse the shipment info list works.

File: src/components/Office/DefinitionLists/ShipmentInfoList.test.jsx [hunk 3]
```diff
@@ -119,7 +123,40 @@ describe('Shipment Info List', () => {
 
     expect(screen.queryByText(labels.secondaryPickupAddress)).toBeNull();
     expect(screen.queryByText(labels.secondaryDeliveryAddress)).toBeNull();
-    expect(screen.queryByText(labels.agents[0])).toBeNull();
-    expect(screen.queryByText(labels.agents[1])).toBeNull();
+    expect(screen.queryByText(labels.mtoAgents[0])).toBeNull();
```
Comment (YanZ777) [inline src/components/Office/DefinitionLists/ShipmentInfoList.test.jsx:126]:
```suggestion
    expect(await screen.queryByText(labels.mtoAgents[0])).not.toBeInTheDocument();
```
Can we use the above format instead for checking that something is not in the document? I find it reads a bit easier and is the preferred method for these types of checks.

**info:** [This document](https://transcom.github.io/mymove-docs/docs/frontend/testing/writing-tests-using-react-testing-library-and-jest) goes into more detail about preferred strategies when writing tests.

File: src/components/Office/DefinitionLists/ShipmentInfoList.test.jsx [hunk 4]
```diff
@@ -117,9 +121,42 @@ describe('Shipment Info List', () => {
...
+      }),
+    ).toBeInTheDocument();
+
     expect(screen.queryByText(labels.secondaryDeliveryAddress)).toBeNull();
-    expect(screen.queryByText(labels.agents[0])).toBeNull();
-    expect(screen.queryByText(labels.agents[1])).toBeNull();
+
+    expect(screen.queryByText(labels.mtoAgents[0])).toBeNull();
+
+    expect(screen.queryByText(labels.mtoAgents[1])).toBeNull();
```
Comment (YanZ777) [inline src/components/Office/DefinitionLists/ShipmentInfoList.test.jsx:154]: **suggestion:** I think these two lines got missed when switching over to using `not.toBeInTheDocument()`.

**Review by YanZ777 — APPROVED** [review]
Besides the comment on the tests, the rest of this looks good to me!

**Comment by cosburn** [comment]
Looks like the expanding functionality is working! Looks like there's some fields missing from the expanded view, here's a list of what should be there in each state --

Visible when collapsed:
- Requested pickup date
- Origin address
- Destination address
- Counselor remarks
- Customer remarks

Visible when expanded:
- Requested pickup date
- Origin address
- Second pickup address
- Releasing agent
- Requested delivery date
- Destination address
- Second delivery address
- Receiving agent
- Customer remarks
- Counselor remarks

**Review by Ryan-Koch — APPROVED** [review]

**Review by cosburn — APPROVED** [review]
Looks good!

**Pull request merged**